src/CoreServices/FSEvents.cs

```diff
@@ -164,6 +165,88 @@ internal FSEventStreamEventsArgs (FSEvent [] events)
 		}
 	}
 
+	/// <summary>
+	/// Creation options for <see cref="FSEventStream"/>.
+	/// </summary>
+	public sealed class FSEventStreamCreateOptions
+	{
+		/// <summary>
+		/// The allocator to use to allocate memory for the stream. If <c>null</c>, the default
+		/// allocator will be used.
+		/// </summary>
+		public CFAllocator? Allocator { get; set; }
+
+		/// <summary>
+		/// A <c>dev_t</c> corresponding to the device which you want to receive notifications from.
+		/// The <c>dev_t</c> is the same as the <c>st_dev</c> field from a <c>stat</c> structure of a
+		/// file on that device or the <c>f_fsid[0]</c> field of a <c>statfs</c> structure.
+		/// </summary>
+		public ulong? DeviceToWatch { get; set; }
+
+		/// <summary>
+		/// A list of directory paths, signifying the root of a filesystem hierarchy to be watched
+		/// for modifications. If <see cref="DeviceToWatch"/> is set, the list of paths should be
+		/// relative to the root of the device. For example, if a volume "MyData" is mounted at
+		/// "/Volumes/MyData" and you want to watch "/Volumes/MyData/Pictures/July", specify a path
+		/// string of "Pictures/July". To watch the root of a volume pass a path of "" (the empty string).
+		/// </summary>
+		public IReadOnlyList<string>? PathsToWatch { get; set; }
+
+		// NB. to be set only by the [Obsolete] FSEventStream .ctors
+		internal NSArray? NSPathsToWatch { get; set; }
+
+		/// <summary>
+		/// The service will supply events that have happened after the given event ID. To ask for
+		/// events "since now," set to <c>null</c> or <see cref="FSEvent.SinceNowId"/>. Often, clients
+		/// will supply the highest-numbered event ID they have received in a callback, which they can
+		/// obtain via <see cref="FSEventStream.LatestEventId">. Do set to zero, unless you want to
+		/// receive events for every directory modified since "the beginning of time" -- an unlikely scenario.
+		/// </summary>
+		public ulong? SinceWhenId { get; set; }
+
+		/// <summary>
+		/// The amount of time the service should wait after hearing about an event from the kernel
+		/// before passing it along to the client via its callback. Specifying a larger value may result
+		/// in more effective temporal coalescing, resulting in fewer callbacks.
+		/// </summary>
+		public TimeSpan Latency { get; set; }
+
+		/// <summary>
+		/// Flags that modify the behavior of the stream being created.
+		/// See <see cref="FSEventStreamCreateFlags"/>.
+		/// </summary>
+		public FSEventStreamCreateFlags Flags { get; set; }
+
+		public FSEventStreamCreateOptions ()
+		{
+		}
+
+		public FSEventStreamCreateOptions (
```

Comment:
Nit - We tends to put all params on the same line unless they are absurdly long, which shouldn't be the case here.